src/main/java/com/artipie/rpm/pkg/ModifiableMetadata.java

```diff
@@ -46,32 +48,41 @@
     /**
      * Old metadata file from modifiable repository.
      */
-    private final Path old;
+    private final PrecedingMetadata preceding;
+
+    /**
+     * Packages count.
+     */
+    private final AtomicLong cnt;
 
     /**
      * Ctor.
      * @param origin Origin
-     * @param old Old metadata from existing repository
+     * @param preceding Old metadata from existing repository
      */
-    public ModifiableMetadata(final Metadata origin, final Path old) {
+    public ModifiableMetadata(final Metadata origin, final PrecedingMetadata preceding) {
         this.origin = origin;
-        this.old = old;
+        this.preceding = preceding;
+        this.cnt = new AtomicLong(0);
     }
 
     @Override
     public void brush(final List<String> pkgs) throws IOException {
-        new XmlMetaJoin(this.origin.output().tag())
-            .merge(this.origin.output().file(), this.old);
+        final Optional<Path> old = this.preceding.findAndUnzip();
+        if (old.isPresent()) {
+            new XmlMetaJoin(this.origin.output().tag())
+                .merge(this.origin.output().file(), old.get());
```

Comment:
@olenagerasimova extract `old.get()` to a variable to avoid calling it twice

Reply:
@Vatavuk I've removed line to delete the file as files creation and cleaning should be handled on the upper level and should not, in my opinion, be this class responsibility. This file is created in the temp directory created to update rpm repo and will be removed when update is finished along with the directory.